### DISCUSSION TITLE
Location App Idea for Future Application

### DISCUSSION THREAD

**Discussion by codetricity** [caitlynbakery/12_take_save_display#2] [Ideas]
# SC2 Interval Shooting 12

* Use RICOH THETA SC2

* UI to set number of pictures to be taken in sequence (example: take 5, 10, 100 pictures with single button press on mobile app)
use [camera.takePicture](https://api.ricoh/docs/theta-web-api-v2.1/commands/camera.take_picture/) to take another picture as soon as the camera is ready
* use [commands.status](https://api.ricoh/docs/theta-web-api-v2.1/protocols/commands_status/) to check when camera is ready for next picture
* display how many shots out of sequence is completed (example: 7/10)
between each shot, get GPS information from onboard phone GPS using [location](https://pub.dev/packages/location) package
    * NOTE: the THETA SC2 does not have GPS inside the camera and must get the GPS information from the mobile phone. The THETA X has GPS inside the camera

* using dart, convert the GPS information from the location package into the format needed for the RICOH THETA camera. The format is a bit confusing. See [this discussion](https://community.theta360.guide/t/updated-in-2022-examples-of-setting-gps-info/1109/8?u=craig) for an example string.
* use RICOH THETA API [gpsInfo](https://api.ricoh/docs/theta-web-api-v2.1/options/gps_info/) option to set the GPS information of the SC2 prior to each shot. Use [camera.setOptions](https://api.ricoh/docs/theta-web-api-v2.1/commands/camera.set_options/) to accomplish this
* app should only operate in still image mode. if the app is not in image mode, have a way to switch it into still image mode if it is accidentally in video mode
* have settings screen for image settings
NOTE: the SC2 cannot set aperture. That setting is only for the Z1
[Exposure Program](https://api.ricoh/docs/theta-web-api-v2.1/options/exposure_program/) - 1, 2, 4, 9
[shutterSpeed](https://api.ricoh/docs/theta-web-api-v2.1/options/shutter_speed/) - I'm not sure if this is a number or string - considering testing the values with Insomnia
[iso](https://api.ricoh/docs/theta-web-api-v2.1/options/iso/)

### Other Reference and Ideas
* this [discussion in community](https://community.theta360.guide/t/how-to-do-interval-shooting-in-changing-lighting-conditions/8157?u=craig) was the impetus for the idea for this project
to test the GPS information, consider using the RICOH THETA official mobile app and then set the GPS information with the app. After you set the GPS info on a picture (just take the picture with the official mobile app), then use your app or camera.listFiles (with Insomnia) to get the fileUrl of the last image taken. Then, pull the string for the GPS with getOptions. Once you have the string, use the same string to for setOptions (since you know that the format is the same)

**Comment by jcasman** [comment]
Based on office discussion, we're building an education version first, to explain the process and work through the steps one-by-one. The audience is a student of the RICOH THETA API. Not focused on being a pretty demo.

# App name: SC2 GPS Tutorial

* First Button: get GPS information from onboard phone GPS using [location](https://pub.dev/packages/location) package
      * NOTE: the THETA SC2 does not have GPS inside the camera and must get the GPS information from the mobile phone.  The THETA X has GPS inside the camera

* Second Button: using dart, convert the GPS information from the location package into the format needed for the RICOH THETA camera.  The format is a bit confusing.  See [this discussion](https://community.theta360.guide/t/updated-in-2022-examples-of-setting-gps-info/1109/8?u=craig) for an example string. 
* Display GPS information on screen

* Third Button: use RICOH THETA API [gpsInfo](https://api.ricoh/docs/theta-web-api-v2.1/options/gps_info/) option to set the GPS information of the SC2 prior to each shot.  Use [camera.setOptions](https://api.ricoh/docs/theta-web-api-v2.1/commands/camera.set_options/) to accomplish this
* SC2 API for metadata is not supported. Workaround needed. Saving picture to phone and confirming GPS info there may work.
* Display confirmation that GPS information is set

* Fourth Button: Take picture

* Fifth Button: Save image to gallery

* Sixth Button: Check metadata of saved image using Flutter package called metadata

## Other Reference and Ideas

* this [discussion in community](https://community.theta360.guide/t/how-to-do-interval-shooting-in-changing-lighting-conditions/8157?u=craig) was the impetus for the idea for this project
* to test the GPS information, consider using the RICOH THETA official mobile app and then set the GPS information with the app.  After  you set the GPS info on a picture (just take the picture with the official mobile app), then use your app or  camera.listFiles (with Insomnia) to get the fileUrl of the last image taken.  Then, pull the string for the GPS with getOptions.  Once you have the string, use the same string to for setOptions (since you know that the format is the same)

**Reply by codetricity** [reply]
set a name for the app and number so that we distinguish it from the previous app I proposed.
put diagram into discussion

**Comment by codetricity** [comment]
# SC2 GPS API Tutorial

This project illustrates how to acquire GPS information from Android and iOS devices and record the information into the RICOH THETA image files.  The example focuses on the RICOH THETA SC2, the lowest-cost model. It should also work with the RICOH THETA V, Z1 and X models.  However, as the RICOH THETA X has built-in GPS, the technique explained in this tutorial is not recommended for the RICOH THETA X.

![Screen Shot 2022-08-03 at 3 34 44 PM](https://user-images.githubusercontent.com/5799860/182739246-22bd0a7a-a74a-43c1-86c8-67813a826a19.png)

It does not appear that the RICOH THETA SC2 supports the [camera._getMetadata ](https://api.ricoh/docs/theta-web-api-v2.1/commands/camera._get_metadata/) API command.  This tutorial shows a workaround, which involves downloading the SC2 image from the camera to the local mobile device and then using local Flutter packages.

## Packages to Evaluate

* [exif](https://pub.dev/packages/exif)
* [metadata](https://pub.dev/packages/metadata)
* [flutter_media_metadata](https://pub.dev/packages/flutter_media_metadata)
